app/Http/Controllers/Widgets/WidgetController.php

```diff
@@ -63,11 +63,17 @@ public function __invoke(Request $request)
 
         if ($this->show_settings) {
             $view = $this->getSettingsView($request);
-        } else {
-            $view = $this->getView($request);
         }
 
         $settings = $this->getSettings();
+
+        if (!$this->show_settings) {
+            if (array_key_exists('device_group', $settings) && !empty($settings['device_group'])) {
```

Comment:
!empty() is satisfactory here, it is always false if array_key_exists() is false.

Reply:
changed